streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsTest.java

```diff
@@ -0,0 +1,283 @@
+        verifyValueSensor(metricNamePrefix, description, RocksDBMetrics::numberOfOpenFilesSensor);
+    }
+
+    @Test
+    public void shouldGetNumberOfFilesErrors() {
+        final String metricNamePrefix = "number-file-errors";
+        final String description = "Total number of file errors occurred";
+        setupStreamsMetricsMock(metricNamePrefix);
+        StreamsMetricsImpl.addTotalMetricToSensor(sensor, STATE_LEVEL_GROUP, tags, metricNamePrefix, description);
+
+        replayCallAndVerify(RocksDBMetrics::numberOfFileErrorsSensor);
+    }
+
+    private void verifyRateAndTotalSensor(final String metricNamePrefix,
+                                          final String descriptionOfTotal,
+                                          final String descriptionOfRate,
+                                          final SensorCreator sensorCreator) {
+        setupStreamsMetricsMock(metricNamePrefix);
+        StreamsMetricsImpl.addAmountRateAndTotalMetricsToSensor(
+            sensor,
+            STATE_LEVEL_GROUP,
+            tags,
+            metricNamePrefix,
+            descriptionOfRate,
+            descriptionOfTotal
+        );
+
+        replayCallAndVerify(sensorCreator);
+    }
+
+    private void verifyRateSensor(final String metricNamePrefix,
+                                  final String description,
+                                  final SensorCreator sensorCreator) {
+        setupStreamsMetricsMock(metricNamePrefix);
+        StreamsMetricsImpl.addAmountRateMetricToSensor(sensor, STATE_LEVEL_GROUP, tags, metricNamePrefix, description);
+
+        replayCallAndVerify(sensorCreator);
+    }
+
+    private void verifyValueSensor(final String metricNamePrefix,
+                                   final String description,
+                                   final SensorCreator sensorCreator) {
+        setupStreamsMetricsMock(metricNamePrefix);
+        StreamsMetricsImpl.addValueMetricToSensor(sensor, STATE_LEVEL_GROUP, tags, metricNamePrefix, description);
+
+        replayCallAndVerify(sensorCreator);
+    }
+
+    private void setupStreamsMetricsMock(final String metricNamePrefix) {
+        mockStatic(StreamsMetricsImpl.class);
+        expect(streamsMetrics.storeLevelSensor(
+            taskName,
+            storeName,
+            metricNamePrefix,
+            RecordingLevel.DEBUG
+        )).andReturn(sensor);
+        expect(streamsMetrics.storeLevelTagMap(
+            taskName,
+            STORE_TYPE_PREFIX + storeType,
+            storeName
+        )).andReturn(tags);
+    }
+
+    private void replayCallAndVerify(final SensorCreator sensorCreator) {
+        replayAll();
+        replay(StreamsMetricsImpl.class);
+
+        final Sensor sensor =
+            sensorCreator.sensor(streamsMetrics, new RocksDBMetricContext(taskName, storeType, storeName));
+
+        verifyAll();
+        verify(StreamsMetricsImpl.class);
+
+        assertThat(sensor, is(this.sensor));
```

Comment:
Hey, I'm sorry, but can you explain what's going on here? `Sensor` doesn't override `equals`, so I'm not seeing how this assertion works.

Reply:
Here I am just verifying that the methods in `RocksDBMetrics` pass through the sensor they get from the methods in the `StreamsMetricImpl` mock. It is an equality check on the reference. Before verifying the sensor, I also verify through the mock whether the metrics with the correct names and tags are added to the sensor.

Reply:
Oh, right, I forgot that the metrics registry returns the same copy of the sensor when all the name, description, and tags are the same... Thanks.